### PR TITLE
Added support for skipping prefixed domain

### LINKED ISSUE #83: Allow absolute messages in PrefixedTranslator using "//"
This is just idea about behavior, when using prefixed translator, snippet of code will say more then 1000 words:

If you agree with this, i can send PR and cover with tests.
```php
$prefixedTranslator = $translator->domain("some.domain");
$prefixedTranslator->translate("//namespace.xx"); // result $translator->translate("namespace.xx")

// What i have to do is:
$prefixedTranslator->unwrap()->translate("namespace.xx");
```

There is 1 very good usecase, i wanted to use global translation required message in config.neon:
```
forms:
	messages:
		FILLED: "forms.required"
```
But afterwards i can not use prefixed translator in forms like this:
```php
$form->setTranslator($this->translator->domain("forms.signIn");
$input->setRequired(); // Result is $translator->translate("forms.signIn.forms.required") instead of just "form.required"
```

### PR DESCRIPTION
**Pull request by akadlec**
If you create translator with domain and want to use translation string from other domain, it's impossible. Now when you put "../" on the begening, translator will skip defined prefix.

For me it is usable when i set domain for module component and want to use one string from framework.

[Closes #83]

### CODE REVIEW

File: src/Kdyby/Translation/PrefixedTranslator.php [hunk 1]
```diff
@@ -49,6 +50,14 @@ public function __construct($prefix, ITranslator $translator)
 
 	public function translate($message, $count = NULL, $parameters = array(), $domain = NULL, $locale = NULL)
 	{
+		if (Utils\Strings::startsWith($message, '../')) {
```
Comment (fprochazka) [inline src/Kdyby/Translation/PrefixedTranslator.php]: I don't think I wanna support relative paths. I would be okey with supporting absolute paths. Something like.

```php
$prefixed->translate('//front.homepage.hello');
```

also, we don't need to check for `../` which are all single-byte characters with multibyte functions. plain old `stripos()` would be enough

Comment (akadlec) [inline src/Kdyby/Translation/PrefixedTranslator.php]: mabye i write it wrong. it is not relative but absolute. The point is to skip the prefix when you use prefixed translator

Comment (akadlec) [inline src/Kdyby/Translation/PrefixedTranslator.php]: is it now ok for you?

**Comment by fprochazka** [comment]
@akadlec Now it looks good :) Can you please rebase it (and solve the conflict), so it can be merged and also add a testcase?

**Comment by akadlec** [comment]
Conflicts should be solved now.

**Comment by fprochazka** [comment]
@akadlec the conflicts are not solved :-/

File: src/Kdyby/Translation/PrefixedTranslator.php [hunk 2]
```diff
@@ -72,68 +90,55 @@ public function unwrap()
 
 
 	/**
-	 * @param Nette\Templating\Template $template
+	 * @param $template
 	 * @return ITranslator
 	 */
-	public function unregister(Nette\Templating\Template $template)
```
Comment (fprochazka) [inline src/Kdyby/Translation/PrefixedTranslator.php]: Also, this seems like a nice addition, but it's unrelated to the original PR. Can you please separate this to another PR?

File: src/Kdyby/Translation/PrefixedTranslator.php [hunk 3]
```diff
@@ -49,6 +49,24 @@ public function __construct($prefix, ITranslator $translator)
 
 	public function translate($message, $count = NULL, $parameters = array(), $domain = NULL, $locale = NULL)
 	{
+		if ($message instanceof Phrase) {
+			$translationString = $message->message;
```
Comment (fprochazka) [inline src/Kdyby/Translation/PrefixedTranslator.php]: imho ternary operator would be simpler

File: src/Kdyby/Translation/PrefixedTranslator.php [hunk 4]
```diff
@@ -49,6 +49,24 @@ public function __construct($prefix, ITranslator $translator)
...
+		if (strpos($translationString, '//') !== FALSE) {
+			$prefix = NULL;
+			$message = Nette\Utils\Strings::substring($translationString, 2);
+
+		} else {
+			$prefix = $this->prefix . '.';
+		}
+
+		if ($message instanceof Phrase) {
+			return $this->translator->translate(new Phrase($prefix . '.' . $translationString, $message->count, $message->parameters, $message->domain, $message->locale));
```
Comment (fprochazka) [inline src/Kdyby/Translation/PrefixedTranslator.php]: should the dot still be here?

**Comment by fprochazka** [comment]
@akadlec also, can you please write some tests? This looks like a bug https://github.com/Kdyby/Translation/pull/88#discussion-diff-41100222, that should be covered with test.
